From 685a3d4c529170b30afe3945c3c48caff48cab0e Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 26 Aug 2018 18:07:08 +0100
Subject: [PATCH 14/14] Fix cross-compilation on Debian-based distros

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 94396bf324..eb7a6b14c8 100644
--- a/setup.py
+++ b/setup.py
@@ -460,7 +460,8 @@ class PyBuildExt(build_ext):
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         if cross_compiling:
             self.add_gcc_paths()
-        self.add_multiarch_paths()
+        else:
+            self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
-- 
2.18.0

