--- Makefile.in.orig	2016-07-04 13:26:14.000000000 -0500
+++ Makefile.in	2016-07-04 13:27:24.000000000 -0500
@@ -242,7 +242,7 @@
 			else	true; \
 			fi; \
 		done
-		$(INSTALL_PROGRAM) python$(EXE) $(BINDIR)/python$(VERSION)$(EXE)
+		$(INSTALL_PROGRAM) Python/python$(EXE) $(BINDIR)/python$(VERSION)$(EXE)
 		if test -f libpython$(VERSION).so; then \
 			$(INSTALL_DATA) libpython$(VERSION).so $(LIBDIR); \
 		else	true; \
@@ -326,15 +326,15 @@
 		done
 		$(INSTALL_DATA) $(srcdir)/LICENSE $(LIBDEST)/LICENSE.txt
 		PYTHONPATH=$(LIBDEST) \
-			./python$(EXE) -tt $(LIBDEST)/compileall.py $(LIBDEST)
+			./Python/python$(EXE) -tt $(LIBDEST)/compileall.py $(LIBDEST)
 		PYTHONPATH=$(LIBDEST) \
-			./python$(EXE) -O $(LIBDEST)/compileall.py $(LIBDEST)
+			./Python/python$(EXE) -O $(LIBDEST)/compileall.py $(LIBDEST)
 
 # Create the PLATDIR source directory, if one wasn't distributed..
 $(srcdir)/Lib/$(PLATDIR):
 		mkdir $(srcdir)/Lib/$(PLATDIR)
 		cp $(srcdir)/Lib/plat-generic/regen $(srcdir)/Lib/$(PLATDIR)/regen
-		export PATH; PATH="`pwd`:$$PATH"; \
+		export PATH; PATH="`pwd`/Python:$$PATH"; \
 		export PYTHONPATH; PYTHONPATH="`pwd`/Lib"; \
 		export EXE; EXE="$(EXE)"; \
 		cd $(srcdir)/Lib/$(PLATDIR); ./regen
