--- Makefile.in.orig	2016-07-02 11:14:59.000000000 -0500
+++ Makefile.in	2016-07-02 11:33:23.000000000 -0500
@@ -188,7 +188,7 @@
 			else	true; \
 			fi; \
 		done
-		$(INSTALL_PROGRAM) python $(BINDIR)/python$(VERSION)
+		$(INSTALL_PROGRAM) Python/python $(BINDIR)/python$(VERSION)
 
 # Install the manual page
 maninstall:
@@ -262,13 +262,13 @@
 			done; \
 		done
 		PYTHONPATH=$(LIBDEST) \
-			./python $(LIBDEST)/compileall.py $(LIBDEST)
+			./Python/python $(LIBDEST)/compileall.py $(LIBDEST)
 
 # Create the MACHDEP source directory, if one wasn't distributed..
 $(srcdir)/Lib/$(MACHDEP):
 		mkdir $(srcdir)/Lib/$(MACHDEP)
 		cp $(srcdir)/Lib/generic/regen $(srcdir)/Lib/$(MACHDEP)/regen
-		export PATH; PATH="`pwd`:$$PATH"; \
+		export PATH; PATH="`pwd`:`pwd`/Python:$$PATH"; \
 		export PYTHONPATH; PYTHONPATH="`pwd`/Lib"; \
 		cd $(srcdir)/Lib/$(MACHDEP); ./regen
 
