From 74b9f6ba7b7205198939f732429b066bf2c2831d Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Wed, 7 Oct 2020 10:08:30 -0500
Subject: [PATCH 10/23] Fix TZPATH on windows

---
 Lib/sysconfig.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index 7dd63b93554..37103c888d1 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -666,6 +666,7 @@ def _init_config_vars():
     if os.name == 'nt':
         _init_non_posix(_CONFIG_VARS)
         _CONFIG_VARS['VPATH'] = sys._vpath
+        _CONFIG_VARS['TZPATH'] = os.path.join(_PREFIX, "share", "zoneinfo")
     if os.name == 'posix':
         _init_posix(_CONFIG_VARS)
     if _HAS_USER_BASE:
