diff -Naur Python-3.9.0rc2.orig/PC/getpathp.c Python-3.9.0rc2/PC/getpathp.c
--- Python-3.9.0rc2.orig/PC/getpathp.c	2020-09-16 18:23:13.000000000 -0300
+++ Python-3.9.0rc2/PC/getpathp.c	2020-09-18 12:53:14.699014326 -0300
@@ -781,9 +781,17 @@
     int skiphome = calculate->home==NULL ? 0 : 1;
 #ifdef Py_ENABLE_SHARED
     if (!Py_IgnoreEnvironmentFlag) {
-        calculate->machine_path = getpythonregpath(HKEY_LOCAL_MACHINE,
-                                                   skiphome);
-        calculate->user_path = getpythonregpath(HKEY_CURRENT_USER, skiphome);
+        char * allow_registry_paths = getenv("CONDA_PY_ALLOW_REG_PATHS");
+        if (allow_registry_paths && allow_registry_paths[0] != '0')
+        {
+            calculate->machine_path = getpythonregpath(HKEY_LOCAL_MACHINE, skiphome);
+            calculate->user_path = getpythonregpath(HKEY_CURRENT_USER, skiphome);
+        }
+        else
+        {
+            calculate->machine_path = NULL;
+            calculate->user_path = NULL;
+        }
     }
 #endif
     /* We only use the default relative PYTHONPATH if we haven't
