From e73c1dc6bdd4d773aa2df6e5a7684e3409f1ab08 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 12 Feb 2020 21:35:50 +0100
Subject: [PATCH 21/27] egg debugging with Windows debug builds

---
 Lib/zipimport.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Lib/zipimport.py b/Lib/zipimport.py
index c55fec6aa1..d90df3e07e 100644
--- a/Lib/zipimport.py
+++ b/Lib/zipimport.py
@@ -513,6 +513,7 @@ def _read_directory(archive):
             path = _bootstrap_external._path_join(archive, name)
             t = (path, compress, data_size, file_size, file_offset, time, date, crc)
             files[name] = t
+            _bootstrap._verbose_message('zipimport: name : {}', name)
             count += 1
     _bootstrap._verbose_message('zipimport: found {} names in {!r}', count, archive)
     return files
-- 
2.30.2

