From f38aa32e3f9bf268ebcf7a6f34446fc7b7e8b32a Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 20 Aug 2021 10:23:51 -0700
Subject: [PATCH 21/26] Unvendor tcltk

---
 PCbuild/tcltk.props | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/PCbuild/tcltk.props b/PCbuild/tcltk.props
index cd54b2567b2..0507b044ee8 100644
--- a/PCbuild/tcltk.props
+++ b/PCbuild/tcltk.props
@@ -17,13 +17,11 @@
     <TixMinorVersion>$([System.Version]::Parse($(TixVersion)).Minor)</TixMinorVersion>
     <TixPatchLevel>$([System.Version]::Parse($(TixVersion)).Build)</TixPatchLevel>
     <TixRevision>$([System.Version]::Parse($(TixVersion)).Revision)</TixRevision>
-    <tclDir Condition="$(tclDir) == ''">$(ExternalsDir)tcl-core-$(TclVersion)\</tclDir>
-    <tkDir Condition="$(tkDir) == ''">$(ExternalsDir)tk-$(TkVersion)\</tkDir>
-    <tixDir Condition="$(tixDir) == ''">$(ExternalsDir)tix-$(TixVersion)\</tixDir>
-    <tcltkDir Condition="$(tcltkDir) == ''">$(ExternalsDir)tcltk-$(TclVersion)\$(ArchName)\</tcltkDir>
-    <tclWin32Exe Condition="$(Platform) == 'Win32'">$(tcltkDir)\bin\tclsh$(TclMajorVersion)$(TclMinorVersion)t.exe</tclWin32Exe>
-    <tclWin32Exe Condition="$(Platform) != 'Win32'">$(tcltkDir)\..\win32\bin\tclsh$(TclMajorVersion)$(TclMinorVersion)t.exe</tclWin32Exe>
-
+    <tclDir>$(condaDir)</tclDir>
+    <tkDir>$(condaDir)</tkDir>
+    <tixDir>$(condaDir)</tixDir>
+    <tcltkDir>$(condaDir)</tcltkDir>
+    <tclWin32Exe>$(condaDir)\bin\tclsh$(TclMajorVersion)$(TclMinorVersion)t.exe</tclWin32Exe>
     <!--<TclDebugExt Condition="'$(Configuration)' == 'Debug'">g</TclDebugExt>-->
     <tclDLLName>tcl$(TclMajorVersion)$(TclMinorVersion)t$(TclDebugExt).dll</tclDLLName>
     <tclLibName>tcl$(TclMajorVersion)$(TclMinorVersion)t$(TclDebugExt).lib</tclLibName>
