From 81a665c32708446040a8e7027f5cd055d3a3b5f5 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 5 Sep 2023 21:51:31 +0200
Subject: [PATCH 21/28] Override configure LIBFFI

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 3499d97f59e..9dfc1191b20 100755
--- a/configure
+++ b/configure
@@ -15929,7 +15929,7 @@ if test "x$ac_cv_lib_ffi_ffi_call" = xyes
 then :
 
                 have_libffi=yes
-        LIBFFI_CFLAGS="-I${SDKROOT}/usr/include/ffi -DUSING_APPLE_OS_LIBFFI=1"
+        LIBFFI_CFLAGS="-I${PREFIX}/include -DUSING_APPLE_OS_LIBFFI=0"
         LIBFFI_LIBS="-lffi"
 
 fi
