From 0295cb4b98ae3e9f0c975c10602d70593db5f1e5 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 5 Sep 2023 21:51:31 +0200
Subject: [PATCH 21/25] Override configure LIBFFI

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index ccb53c64154..ac9d56757cd 100755
--- a/configure
+++ b/configure
@@ -15248,7 +15248,7 @@ if test "x$ac_cv_lib_ffi_ffi_call" = xyes
 then :
 
                 have_libffi=yes
-        LIBFFI_CFLAGS="-I${SDKROOT}/usr/include/ffi -DUSING_APPLE_OS_LIBFFI=1"
+        LIBFFI_CFLAGS="-I${PREFIX}/include -DUSING_APPLE_OS_LIBFFI=0"
         LIBFFI_LIBS="-lffi"
 
 fi
