From 4966a9e99c4ad67ce3b14b37761932e26ea816d0 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 20 Aug 2021 10:23:51 -0700
Subject: [PATCH 22/25] Unvendor tcltk

---
 PCbuild/tcltk.props | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/PCbuild/tcltk.props b/PCbuild/tcltk.props
index 16dc35d45e..f0ae3372f3 100644
--- a/PCbuild/tcltk.props
+++ b/PCbuild/tcltk.props
@@ -14,10 +14,10 @@
     <TixMinorVersion>4</TixMinorVersion>
     <TixPatchLevel>3</TixPatchLevel>
     <TixRevision>6</TixRevision>
-    <tclDir>$(ExternalsDir)tcl-core-$(TclMajorVersion).$(TclMinorVersion).$(TclPatchLevel).$(TclRevision)\</tclDir>
-    <tkDir>$(ExternalsDir)tk-$(TkMajorVersion).$(TkMinorVersion).$(TkPatchLevel).$(TkRevision)\</tkDir>
-    <tixDir>$(ExternalsDir)tix-$(TixMajorVersion).$(TixMinorVersion).$(TixPatchLevel).$(TixRevision)\</tixDir>
-    <tcltkDir>$(ExternalsDir)tcltk-$(TclMajorVersion).$(TclMinorVersion).$(TclPatchLevel).$(TclRevision)\$(ArchName)\</tcltkDir>
+    <tclDir>$(condaDir)</tclDir>
+    <tkDir>$(condaDir)</tkDir>
+    <tixDir>$(condaDir)</tixDir>
+    <tcltkDir>$(condaDir)</tcltkDir>
     <!--<TclDebugExt Condition="'$(Configuration)' == 'Debug'">g</TclDebugExt>-->
     <tclDLLName>tcl$(TclMajorVersion)$(TclMinorVersion)t$(TclDebugExt).dll</tclDLLName>
     <tclLibName>tcl$(TclMajorVersion)$(TclMinorVersion)t$(TclDebugExt).lib</tclLibName>
-- 
2.32.1 (Apple Git-133)

