From b1ceec3f7ca1dbfd89f5201b567819d4eacc9fe9 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 3 Sep 2025 13:32:24 +0200
Subject: [PATCH 25/25] Search LLVM_ROOT for llvm binaries

---
 Tools/jit/_llvm.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Tools/jit/_llvm.py b/Tools/jit/_llvm.py
index b68da1b6be1..88070133ddb 100644
--- a/Tools/jit/_llvm.py
+++ b/Tools/jit/_llvm.py
@@ -68,6 +68,11 @@ async def _get_brew_llvm_prefix(*, echo: bool = False) -> str | None:
 
 @_async_cache
 async def _find_tool(tool: str, *, echo: bool = False) -> str | None:
+    # Explicitly defined LLVM installation location
+    if (llvm_root := os.getenv("LLVM_ROOT")) is not None:
+        path = os.path.join(llvm_root, "bin", tool)
+        if await _check_tool_version(path, echo=echo):
+            return path
     # Unversioned executables:
     path = tool
     if await _check_tool_version(path, echo=echo):
