From bd729d83e85f11f7b43d33b508b0f47ac5129269 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 18 Feb 2020 17:38:32 +0100
Subject: [PATCH 26/26] Debug _PyImport_FindSharedFuncptrWindows() values of
 _PyInterpreterState_Get()

---
 Python/dynload_win.c | 50 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)

diff --git a/Python/dynload_win.c b/Python/dynload_win.c
index 56d8913455..0efde6f26e 100644
--- a/Python/dynload_win.c
+++ b/Python/dynload_win.c
@@ -202,13 +202,63 @@ dl_funcptr _PyImport_FindSharedFuncptrWindows(const char *prefix,
            AddDllDirectory function. We add SEARCH_DLL_LOAD_DIR to
            ensure DLLs adjacent to the PYD are preferred. */
         /* This resyncs values in PATH to AddDllDirectory() */
+
+        PyInterpreterState* ts = _PyInterpreterState_Get();
+        if (!ts)
+        {
+            fwprintf(stderr, L"(0) ts = 0");
+        }
+        else
+        {
+            fwprintf(stderr, L"(0) ts is OK");
+        }
+
         extern int CondaEcosystemModifyDllSearchPath(int, int);
         CondaEcosystemModifyDllSearchPath(1, 1);
+        ts = _PyInterpreterState_Get();
+        if (!ts)
+        {
+            fwprintf(stderr, L"(1) ts = 0");
+        }
+        else
+        {
+            fwprintf(stderr, L"(1) ts is OK");
+        }
 
         Py_BEGIN_ALLOW_THREADS
+
+        ts = _PyInterpreterState_Get();
+        if (!ts)
+        {
+            fwprintf(stderr, L"(2) ts = 0");
+        }
+        else
+        {
+            fwprintf(stderr, L"(2) ts is OK");
+        }
         hDLL = LoadLibraryExW(wpathname, NULL,
                               LOAD_WITH_ALTERED_SEARCH_PATH);
+        ts = _PyInterpreterState_Get();
+        if (!ts)
+        {
+            fwprintf(stderr, L"(3) ts = 0");
+        }
+        else
+        {
+            fwprintf(stderr, L"(3) ts is OK");
+        }
+
         Py_END_ALLOW_THREADS
+
+        ts = _PyInterpreterState_Get();
+        if (!ts)
+        {
+            fwprintf(stderr, L"(4) ts = 0");
+        }
+        else
+        {
+            fwprintf(stderr, L"(4) ts is OK");
+        }
 #if HAVE_SXS
         _Py_DeactivateActCtx(cookie);
 #endif
-- 
2.25.0

