From a29ebfcab4451f44872e2651f1d03a54b40a88f3 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 16 Aug 2021 03:07:40 -0700
Subject: [PATCH 27/27] Unvendor libffi

---
 PCbuild/libffi.props | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/PCbuild/libffi.props b/PCbuild/libffi.props
index 975c4a0d35..5700e755ab 100644
--- a/PCbuild/libffi.props
+++ b/PCbuild/libffi.props
@@ -2,15 +2,15 @@
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemDefinitionGroup>
     <ClCompile>
-      <AdditionalIncludeDirectories>$(libffiIncludeDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(condaDir)\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
-      <AdditionalLibraryDirectories>$(libffiOutDir);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
-      <AdditionalDependencies>libffi-7.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(condaDir)\lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalDependencies>ffi.lib;%(AdditionalDependencies)</AdditionalDependencies>
     </Link>
   </ItemDefinitionGroup>
   <ItemGroup>
-    <_LIBFFIDLL Include="$(libffiOutDir)\libffi-7.dll" />
+    <_LIBFFIDLL Include="$(condaDir)\bin\ffi-7.dll" />
   </ItemGroup>
   <Target Name="_CopyLIBFFIDLL" Inputs="@(_LIBFFIDLL)" Outputs="@(_LIBFFIDLL->'$(OutDir)%(Filename)%(Extension)')" AfterTargets="Build">
     <Copy SourceFiles="@(_LIBFFIDLL)" DestinationFolder="$(OutDir)" />
-- 
2.30.2

