From 1b6604b52651f3088fe07dde50cfc8093124f68a Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 1 Sep 2023 23:32:14 +0200
Subject: [PATCH 20/21] Don't checksharedmods

---
 Makefile.pre.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 7acd876bc8..fa0f42cc71 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -611,7 +611,7 @@ all:		@DEF_MAKE_ALL_RULE@
 
 .PHONY: build_all
 build_all:	check-clean-src $(BUILDPYTHON) platform sharedmods \
-		gdbhooks Programs/_testembed scripts checksharedmods rundsymutil
+		gdbhooks Programs/_testembed scripts rundsymutil
 
 .PHONY: build_wasm
 build_wasm: check-clean-src $(BUILDPYTHON) platform sharedmods \
