From afddfa8176cf8969055c6319504069b250723d3d Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 25 Jan 2021 03:28:08 -0600
Subject: [PATCH] Make dyld search work with SYSTEM_VERSION_COMPAT=1

---
 Modules/_ctypes/callproc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/_ctypes/callproc.c b/Modules/_ctypes/callproc.c
index b0f1e0bd04..f382971dbd 100644
--- a/Modules/_ctypes/callproc.c
+++ b/Modules/_ctypes/callproc.c
@@ -1452,7 +1452,7 @@ copy_com_pointer(PyObject *self, PyObject *args)
 #ifdef __APPLE__
 #ifdef HAVE_DYLD_SHARED_CACHE_CONTAINS_PATH
 #define HAVE_DYLD_SHARED_CACHE_CONTAINS_PATH_RUNTIME \
-    __builtin_available(macOS 11.0, iOS 14.0, tvOS 14.0, watchOS 7.0, *)
+    __builtin_available(macOS 10.16, iOS 14.0, tvOS 14.0, watchOS 7.0, *)
 #else
 // Support the deprecated case of compiling on an older macOS version
 static void *libsystem_b_handle;
-- 
2.33.0

